Source/Src_nd/ppm.F90

```diff
@@ -1355,2216 +1354,2219 @@ subroutine ppm_2d(lo,hi,s,s_lo,s_hi,nc_s,&
-         sedge = (7.d0/12.d0)*(s(i,j-1,k)+s(i,j,k)) &
-              - (1.d0/12.d0)*(s(i,j-2,k)+s(i,j+1,k))
-         !
-         ! Limit sedge.
-         if ((sedge-s(i,j-1,k))*(s(i,j,k)-sedge) .lt. ZERO) then
-            D2  = THREE*(s(i,j-1,k)-TWO*sedge+s(i,j,k))
-            D2L = s(i,j-2,k)-TWO*s(i,j-1,k)+s(i,j,k)
-            D2R = s(i,j-1,k)-TWO*s(i,j,k)+s(i,j+1,k)
-            sgn = sign(ONE,D2)
-            D2LIM = sgn*max(min(C*sgn*D2L,C*sgn*D2R,sgn*D2),ZERO)
-            sedge = HALF*(s(i,j-1,k)+s(i,j,k)) - SIXTH*D2LIM
-         end if
-
-         ! +1
-         ! Interpolate s to y-edges.
-         sedger = (7.d0/12.d0)*(s(i,j,k)+s(i,j+1,k)) &
-              - (1.d0/12.d0)*(s(i,j-1,k)+s(i,j+2,k))
-         !
-         ! Limit sedge.
-         if ((sedger-s(i,j,k))*(s(i,j+1,k)-sedger) .lt. ZERO) then
-            D2  = THREE*(s(i,j,k)-TWO*sedger+s(i,j+1,k))
-            D2L = s(i,j-1,k)-TWO*s(i,j,k)+s(i,j+1,k)
-            D2R = s(i,j,k)-TWO*s(i,j+1,k)+s(i,j+2,k)
-            sgn = sign(ONE,D2)
-            D2LIM = sgn*max(min(C*sgn*D2L,C*sgn*D2R,sgn*D2),ZERO)
-            sedger = HALF*(s(i,j,k)+s(i,j+1,k)) - SIXTH*D2LIM
-         end if
-
-         ! +2
-         ! Interpolate s to y-edges.
-         sedgerr = (7.d0/12.d0)*(s(i,j+1,k)+s(i,j+2,k)) &
-              - (1.d0/12.d0)*(s(i,j,k)+s(i,j+3,k))
-         !
-         ! Limit sedge.
-         if ((sedgerr-s(i,j+1,k))*(s(i,j+2,k)-sedgerr) .lt. ZERO) then
-            D2  = THREE*(s(i,j+1,k)-TWO*sedgerr+s(i,j+2,k))
-            D2L = s(i,j,k)-TWO*s(i,j+1,k)+s(i,j+2,k)
-            D2R = s(i,j+1,k)-TWO*s(i,j+2,k)+s(i,j+3,k)
-            sgn = sign(ONE,D2)
-            D2LIM = sgn*max(min(C*sgn*D2L,C*sgn*D2R,sgn*D2),ZERO)
-            sedgerr = HALF*(s(i,j+1,k)+s(i,j+2,k)) - SIXTH*D2LIM
-         end if
-
-        !
-        ! Use Colella 2008 limiters.
-        ! This is a new version of the algorithm
-        ! to eliminate sensitivity to roundoff.
-
-         alphap = sedger-s(i,j,k)
-         alpham = sedge-s(i,j,k)
-         bigp = abs(alphap).gt.TWO*abs(alpham)
-         bigm = abs(alpham).gt.TWO*abs(alphap)
-         extremum = .false.
-
-         if (alpham*alphap .ge. ZERO) then
-            extremum = .true.
-         else if (bigp .or. bigm) then
-            !
-            ! Possible extremum. We look at cell centered values and face
-            ! centered values for a change in sign in the differences adjacent to
-            ! the cell. We use the pair of differences whose minimum magnitude is the
-            ! largest, and thus least susceptible to sensitivity to roundoff.
-            !
-            dafacem = sedge - sedgel
-            dafacep = sedgerr - sedger
-            dabarm = s(i,j,k) - s(i,j-1,k)
-            dabarp = s(i,j+1,k) - s(i,j,k)
-            dafacemin = min(abs(dafacem),abs(dafacep))
-            dabarmin= min(abs(dabarm),abs(dabarp))
-            if (dafacemin.ge.dabarmin) then
-               dachkm = dafacem
-               dachkp = d
```

Comment:
I don't think we should be relying on the fact that lo(1) is now the old lo(1)-1.  We should refer to either domlo() here or pass in also the size of the valid box (I think we do this in Castro as vlo).  This is done throughout this routine and makes it confusing.